Include/httpClient/task.h

```diff
@@ -37,17 +37,32 @@ typedef HC_RESULT
     _In_opt_ void* completionRoutineContext
     );
 
+/// <summary>
+/// The task event type
+/// </summary>
 typedef enum HC_TASK_EVENT_TYPE
 {
+    /// <summary>
+    /// The task is in a pending task and has not started executing 
+    /// </summary>
     HC_TASK_EVENT_PENDING,
+
+    /// <summary>
+    /// The task has started executing but has not yet completed execution
+    /// </summary>
     HC_TASK_EVENT_EXECUTE_STARTED,
```

Comment:
Just curious when this would be needed?

Reply:
Its not directly needed but may be useful for logging or analytics to monitor the task system